src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -303,7 +303,7 @@ private void GetPackageAndFileDependencies(LockFileTarget target)
 
             var transitiveProjectRefs = new HashSet<string>(
                 target.Libraries
-                    .Where(lib => lib.IsTransitiveProjectReference(LockFile, ref _projectFileDependencies))
+                    .Where(lib => lib.IsTransitiveProjectReference(LockFile, ref _projectFileDependencies, target.TargetFramework.ToString()))
```

Comment:
The TargetFramework here needs to be mapped back to the corresponding framework alias, instead of just calling ToString() on it.  You can do this with `GetLockFileTargetAlias` from `LockFileExtensions`.